x-pack/plugins/enterprise_search/public/applications/app_search/utils/encode_path_params/index.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useParams } from 'react-router-dom';
+
+export const encodePathParams = (pathParams: Record<string, string | number>) => {
+  const encodedParams: Record<string, string> = {};
+
+  Object.entries(pathParams).map(([key, value]) => {
+    encodedParams[key] = encodeURIComponent(value);
+  });
+
+  return encodedParams;
+};
+
+export const useDecodedParams = () => {
+  const decodedParams: Record<string, string> = {};
+
+  const params = useParams();
+  Object.entries(params).map(([key, value]) => {
+    decodedParams[key] = decodeURIComponent(value as string);
+  });
+
+  return decodedParams;
+};
```

Comment:
Just want to note - on some reflection, it's maybe a little overkill to have a helper for 2 instances in our app, but I think it'll be worth it (from the perspective of future-proofing) that we have one documented and canonical place where we ensure our URLs are correctly encoded, vs. doing it manually and maybe slightly differently in multiple places in the future.

Reply:
I agree.